### ISSUE TITLE
[ANTLR] subroutineCall should take a indexIdentifierList instead of expressionList

### ISSUE THREAD

**Issue by hhorii** [Qiskit/openqasm#177]
Currently, `subroutineCall` takes a `expressionList `.

```
subroutineCall
    : Identifier ( LPAREN expressionList? RPAREN )? expressionList
```

I guess that this `expressionList` should be replaced with `indexIdentifierList` as `quantumGateCall` does.

```
quantumGateCall
    : quantumGateModifier* quantumGateName ( LPAREN expressionList? RPAREN )? indexIdentifierList
    ;
```

**Comment by zachschoenfeld33** [comment]
This is correct, thanks @hhorii. Please add a PR with the fix and a test and we can get it merged it.

**Comment by lusvelt** [comment]
I just had an issue related to this, which I state here in case it helps others.
If I try to parse a subroutine which contains
`return a - b;`
it gets parsed as subroutineCall (`a` is the subroutine Identifier and `-b` is the expression passed as an argument).
With the fix you suggested here it gets parsed correctly.
